### PR TITLE
[libcommhistory] Add contactName role to eventmodel. Contributes to JB#56694

### PR DESCRIPTION
**Pull request by aroditelev**
This role may be needed if a developer wants to implement a call history search.

### CODE REVIEW

File: src/eventmodel.h [hunk 1]
```diff
@@ -103,7 +103,8 @@ class LIBCOMMHISTORY_EXPORT EventModel: public QAbstractItemModel
         FromVCardLabel,
         ReadStatus,
         SubscriberIdentity,
-        NumberOfColumns
+        NumberOfColumns,
+        ContactName
```
Comment (pvuorela) [inline src/eventmodel.h]: I think the idea here is that the NumberOfColumns is the last one so it indicates, well, what it says.

Strange setup though with separate roles and columns.

Comment (okodron) [reply src/eventmodel.h]: Indeed, it should be the last one in the list.

Comment (aroditelev) [reply src/eventmodel.h]: I agree, I did not notice that this is really not a role

File: src/eventmodel.cpp [hunk 2]
```diff
@@ -80,6 +80,7 @@ QHash<int, QByteArray> EventModel::roleNames() const
     roles[BaseRole + FromVCardLabel] = "fromVCardLabel";
     roles[BaseRole + ReadStatus] = "readStatus";
     roles[BaseRole + SubscriberIdentity] = "subscriberIdentity";
+    roles[BaseRole + ContactName] = "contactName";
```
Comment (pvuorela) [inline src/eventmodel.cpp]: Ehm, speaking of that role vs column, what the heck has been happening here. From the git history looks like initial commit was having columns and then later also roles got added. Makes me wonder if the initial thing was done wrong and then never properly fixed.

**Review by okodron — APPROVED** [review]

File: src/eventmodel.h (hunk 1 above)
Comment (aroditelev) [reply src/eventmodel.h]: @pvuorela  made changes

File: src/eventmodel.cpp (hunk 2 above)
Comment (aroditelev) [reply src/eventmodel.cpp]: @pvuorela  As far as I understand, the Tree model seems to assume here, if there are columns, then it should display them as columns.

Comment (aroditelev) [reply src/eventmodel.cpp]: And if only the rows, then only they will be displayed, but my opinion is the solution is very strange, it looks sloppy.

**Review by xlise — APPROVED** [review]

File: src/eventmodel.cpp (hunk 2 above)
Comment (pvuorela) [reply src/eventmodel.cpp]: I'd like to get that sorted out, but guess it shouldn't be blocking adding new things.

However for the new thing is there a reason why it's a column and not a role?

Comment (aroditelev) [reply src/eventmodel.cpp]: so that you can get the value from outside ```sourceModel()->index(sourceRow, 0).data(roleId).toString()```

Comment (pvuorela) [reply src/eventmodel.cpp]: Yea but now it's not a roleId, it's a ColumnId. Above might work but really by the api it should be something like index(sourceRow, ContactName).data(/*any integer below BaseRole to work */), right?

Comment (aroditelev) [reply src/eventmodel.cpp]: Yes that's right

Comment (pvuorela) [reply src/eventmodel.cpp]: And still to add, your example above would work more by the book if this new thing was a role instead?

Comment (aroditelev) [reply src/eventmodel.cpp]: At the moment, it is more convenient to get the value from one enum, since, from it, for example, RemoteUid is used

Comment (pvuorela) [reply src/eventmodel.cpp]: Not sure I'm following. So what I'm mostly suggesting is moving the new role name to the latter enum, after DateAndAccountGroupingRole, and then declaring the usage here without any BaseRole extra addition. That is, like the roles are commonly used, avoiding the column strangeness.

Comment (aroditelev) [reply src/eventmodel.cpp]: Done

**Review by pvuorela — APPROVED** [review]

**Pull request merged**